### ISSUE TITLE
Obok not running in Calibre 6.29

### ISSUE THREAD

**Issue by Bozana7** [noDRM/DeDRM_tools#428]
It's not converting at all now, even though it says it's successful dedrm, it's coming out as unreadable gobbly goop when you open the book. I didn't have this problem till this new update/upgrade in Calibre. Do I need to downgrade calibre? Or is it OBok itself?

**Comment by ElleKayEm** [comment]
Still working for me.  Sounds like the book was misidentified as DRM-free.  Was the book showing an open green lock next to it in Obok instead of a closed red one?  I've seen others mention that closing Kobo Desktop before importing with Obok has prevented this from happening.  Which version of Obok?

**Comment by Bozana7** [comment]
![Screenshot 2023-11-09 025542](https://github.com/noDRM/DeDRM_tools/assets/120107379/fed9090b-28f0-4c8d-9ede-bd477ae7a716)
![Screenshot 2023-11-09 025635](https://github.com/noDRM/DeDRM_tools/assets/120107379/5c982600-fac2-4121-85ca-6662e35d8ea4)
as you can see, when I import it, I always have Kobo closed before importing. I even deleted it a number of times and readding it.. just to see. and still no go. It's the newest release book. I've asked friends to try and get it, and they're having the same problem getting the book.

I've using the newest one. Obok v10.0.9

And I'm a Network Egineer. I.T. is my cup of tea ;-)

**Comment by ElleKayEm** [comment]
I don't know.  I can't duplicate the problem.  But I wasn't checking with a brand new release.  Maybe something has changed.  In the past, I have very rarely had this happen, but a restart of both Kobo Desktop and calibre has always resolved it for me.  Even when I leave the Kobo app open, the drmed books correctly show the red closed locks.  If they are incorrectly showing green open locks, Obok just imports them as is without even attempting to decrypt them.  It assumes they are already drm-free.  Are they purchased books or Kobo Plus?  If purchased, as a workaround, you could try downloading the Adobe DRM versions and using regular DeDRM.

**Comment by Bozana7** [comment]
It's a Kobo Plus book...

**Comment by ElleKayEm** [comment]
That may very well be the issue.  Those books may have their DRM set up in a way that Obok doesn't recognize.  Unzip one and see if it has a rights.xml file.

**Comment by ElleKayEm** [comment]
Did a little more digging into this.  It seems that Kobo Plus books work just like Overdrive library books in Kobo Desktop.  It will appear to Obok that there isn't any DRM because there isn't any rights.xml file inside the kepub.  Books that are sold with DRM will still be encrypted though.  If you happen to borrow a book that is normally sold drm-free, those won't be encrypted.  If you actually buy the Kobo Plus book, then you should be able to re-download it and decrypt it with Obok (or with ADE/DeDRM).

**Comment by ElleKayEm** [comment]
Okay, I spoke too soon.  Just tried again with an Overdrive book that was doing what your Kobo Plus book was doing....showing unencrypted in Obok but turning out to be gibberish (still actually encrypted) when added to calibre.  Tried again with the same book a little while later...showed as DRMed in Obok and successfully decrypted when added to calibre.

**Comment by Bozana7** [comment]
but when you open it, is it still gibberish?

**Comment by ElleKayEm** [comment]
No, it showed the red closed lock correctly in Obok, and once added, it was truly decrypted.
